library/runtime/src/commonMain/kotlin/io/matthewnelson/kmp/tor/runtime/internal/ObserverConfChanged.kt

```diff
@@ -31,19 +32,48 @@ internal open class ObserverConfChanged internal constructor(
 ) {
 
     protected override fun notify(data: String) {
-        for (line in data.lines()) { line.parse() }
-    }
+        var network: TorState.Network? = null
+        var socks: LinkedHashSet<String>? = null
+
+        for (line in data.lines()) { when {
+            // DisableNetwork=0
+            // DisableNetwork=1
+            // DisableNetwork  << Implied 1
+            line.startsWith(TorConfig.DisableNetwork.name, ignoreCase = true) -> {
+                network = if (line.substringAfter('=') == "0") {
+                    TorState.Network.Enabled
+                } else {
+                    TorState.Network.Disabled
+                }
+            }
+
+            // __SocksPort=unix:"/tmp/kmp_tor_test/obs_conn_no_net/work/socks2.sock" OnionTrafficOnly GroupWritable
+            // SocksPort=unix:"/tmp/kmp_tor_test/obs_conn_no_net/work/socks3.sock"
+            // SocksPort=unix:/tmp/kmp_tor_test/obs_conn_no_net/work/socks4.sock
+            // __SocksPort=9055
+            // SocksPort=127.0.0.1:9056 OnionTrafficOnly
+            // SocksPort=[::1]:9055
+            line.startsWith(SOCKS_PORT_EPHEMERAL, ignoreCase = true)
+            || line.startsWith(SOCKS_PORT, ignoreCase = true) -> {
+                if (socks == null) {
+                    socks = LinkedHashSet(1, 1.0F)
+                }
 
-    private fun String.parse() = when {
-        startsWith(TorConfig.DisableNetwork.name, ignoreCase = true) -> {
-            val network = if (substringAfter('=') == "0") {
-                TorState.Network.Enabled
-            } else {
-                TorState.Network.Disabled
+                socks.add(line.substringAfter(SOCKS_PORT, ""))
```

Comment:
better to use `'='` as the delimiter b/c `ignoreCase` is true and we already know that there's an `=` because it's included in the `startsWith`

Reply:
Resolved in commit 3befb43